waku/v2/protocol/rln/web3/web3.go

```diff
@@ -26,12 +28,21 @@ type RLNContract struct {
 	DeployedBlockNumber uint64
 }
 
+// EthClientI is an interface for the ethclient.Client, so that we can pass mock client for testing
+type EthClientI interface {
+	bind.ContractBackend
+	TransactionReceipt(ctx context.Context, txHash common.Hash) (*types.Receipt, error)
+	BlockByNumber(ctx context.Context, number *big.Int) (*types.Block, error)
+	SubscribeNewHead(ctx context.Context, ch chan<- *types.Header) (ethereum.Subscription, error)
+	Close()
+}
+
```

Comment:
This naming convention of adding `I` suffix looks too java/csharp-ish. Usually in go we use `-er`. But maybe in this case could it be renamed to `ETHClient` instead?